### PR TITLE
The Great Auxtools Switch

### PR DESCRIPTION
**Pull request by ike709**
Dumps extools, ports auxtools.

Ports https://github.com/Citadel-Station-13/Citadel-Station-13/pull/13991

## Changelog
:cl:
code: Replaced extools with auxtools.
add: Added Putnam's auxmos, which is like TG's atmos but incredibly performant.
del: Removed fastmos, because it's no longer maintained by anybody and it's very unstable.
code: Atmos is now even more performant than fastmos was.
tweak: Firelocks require a crowbar to open/close again.
tweak: Firelocks are fulltile again.
/:cl:

### CODE REVIEW

**Comment by bloons3** [comment]
>Currently causes DD to instantly crash locally

**Comment by ike709** [comment]
>> Currently causes DD to instantly crash locally

> but I had the same issue with extools so it might just be a problem with my PC.

lern 2 reed

**Comment by PowerfulBacon** [comment]
Apparently 514 breaks transparency

**Comment by jupyterkat** [comment]
Crash seems to be caused by /datum/gas_mixture/react(datum/holder). Deleting it makes everything works again on my side. The other issue of the crash is their gas lists work differently and needs Citadel-Station-13/Citadel-Station-13#8236 ported.

**Comment by ike709** [comment]
Putnam says it isn't loading. So I need to figure that out.

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -15,9 +13,10 @@
 			if (ispath(req))
 				var/datum/gas/req_gas = req
 				if (!reaction_key || initial(reaction_key.rarity) > initial(req_gas.rarity))
-					reaction_key = req_gas
-		.[reaction_key] += list(reaction)
-		sortTim(., /proc/cmp_gas_reactions, TRUE)
+					eaction_key = req_gas
```
Comment (ike709) [inline code/modules/atmospherics/gasmixtures/reactions.dm:16]:
```suggestion
					reaction_key = req_gas
```

**Comment by ike709** [comment]
Note to self: Port https://github.com/tgstation/tgstation/pull/59497

Also we're on 514 now so I can rip that out.

**Comment by ike709** [comment]
~~TODO: Fix this runtime~~
image

**Comment by MCterra10** [comment]
beginning map rework for compatibility with auxmos

**Comment by ike709** [comment]
CI is only failing because we haven't pushed an update release on our auxmos fork yet.

**Comment by BrumalLake** [comment]
soon

**Comment by ike709** [comment]
~~TODO: Remove the high pressure woosh graphics~~

**Comment by ike709** [comment]
~~TODO: Remove bumping open firelocks.~~

**Comment by ike709** [comment]
~~TODO: Remove hand-closing firelocks.~~

**Comment by ike709** [comment]
~~TODO: `/obj/machinery/atmospherics/components/trinary/mixer` only uses one of its two inputs.~~

Maybe fixed?

**Comment by ike709** [comment]
~~TODO: Nitrous (N2O) tanks are full of nitryl.~~

**Comment by ike709** [comment]
Now includes poor man's fastmos: https://github.com/Citadel-Station-13/Citadel-Station-13/pull/14954

**Comment by ike709** [comment]
~~TODO: Update auxmos again~~

**Pull request merged**